changelog.d/15498.doc

```diff
@@ -0,0 +1 @@
+Update development docs referencing versions of sqlite we no longer support.
```

Comment:
```suggestion
Update outdated development docs that mention restrictions in versions of SQLite that we no longer support.
```

I don't want to alarm anyone by making this sound like it changes the versions of SQLite we support silently, so I've tried to bring attention to the docs just being outdated.